File: src/main/java/net/greenjab/fixedminecraft/registry/item/map_book/MapBookAdditionsComponent.java
package net.greenjab.fixedminecraft.registry.item.map_book;

import com.mojang.serialization.Codec;
import net.greenjab.fixedminecraft.network.IntArray;
import net.minecraft.network.RegistryFriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import java.util.ArrayList;
import java.util.List;

public record MapBookAdditionsComponent(List<Integer> additions) {
    public static final MapBookAdditionsComponent DEFAULT = new MapBookAdditionsComponent(new ArrayList<>());

    public static final Codec<MapBookAdditionsComponent> CODEC = Codec.INT.listOf().xmap(MapBookAdditionsComponent::new, MapBookAdditionsComponent::additions);
    public static final StreamCodec<RegistryFriendlyByteBuf, MapBookAdditionsComponent> PACKET_CODEC = StreamCodec.composite(
            IntArray.LIST_CODEC,
            MapBookAdditionsComponent::additions,
            MapBookAdditionsComponent::new
    );
}

File: src/main/kotlin/net/greenjab/fixedminecraft/registry/item/map_book/MapBookAdditionsComponent.kt
package net.greenjab.fixedminecraft.registry.item.map_book

import com.mojang.serialization.Codec
import net.greenjab.fixedminecraft.network.IntArray
import net.minecraft.network.RegistryByteBuf
import net.minecraft.network.codec.PacketCodec


@JvmRecord
data class MapBookAdditionsComponent(val additions: List<Int>) {
   /* companion object {
        val DEFAULT: MapBookAdditionsComponent = MapBookAdditionsComponent(ArrayList())

        val CODEC: Codec<MapBookAdditionsComponent> = Codec.INT.listOf().xmap({ additions: List<Int> ->
            MapBookAdditionsComponent(
                additions
            )
        }, MapBookAdditionsComponent::additions)
        val PACKET_CODEC: PacketCodec<RegistryByteBuf, MapBookAdditionsComponent> = PacketCodec.tuple(
            IntArray.LIST_CODEC,
            MapBookAdditionsComponent::additions
        ) { additions: List<Int> ->
            MapBookAdditionsComponent(
                additions
            )
        }
    }*/
}
